src/main/java/core/basesyntax/Engine.java

```diff
@@ -1,5 +1,75 @@
 package core.basesyntax;
 
-public class Engine {
+public class Engine implements Cloneable {
+    private String name;
+    private int horsePower;
+    private int numberOfCylinders;
+    private double volume;
+    private double consumption;
 
+    public double getConsumption() {
+        return consumption;
+    }
+
+    public void setConsumption(double consumption) {
+        this.consumption = consumption;
+    }
+
+    public double getVolume() {
+        return volume;
+    }
+
+    public void setVolume(double volume) {
+        this.volume = volume;
+    }
+
+    public int getNumberOfCylinders() {
+        return numberOfCylinders;
+    }
+
+    public void setNumberOfCylinders(int numberOfCylinders) {
+        this.numberOfCylinders = numberOfCylinders;
+    }
+
+    public int getHorsePower() {
+        return horsePower;
+    }
+
+    public void setHorsePower(int horsePower) {
+        horsePower = horsePower;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public String toString() {
+        return "Engine{"
+                + "name='" + name + '\''
+                + ", HorsePower=" + horsePower
+                + ", numberOfCylinders=" + numberOfCylinders
+                + ", volume=" + volume
+                + ", consumption=" + consumption
+                + '}';
+    }
+
+    @Override
+    public Engine clone() {
+        try {
+            Engine clonedEngine = (Engine) super.clone();
```

Comment:
what is the result of super.clone() before setters?

Reply:
ok, I understood